### PR TITLE
Finish io handler

### PR DESCRIPTION
**Pull request by mdugacek**
Closing #10

### CODE REVIEW

File: untrusted/io_handler.cpp [hunk 1]
```diff
@@ -177,19 +180,41 @@ void io_handler::handle_remove_entry()
...
+        return;
+    }
+    
+    std::cout << "Listing all ids:\n";
+    for (const auto& id : enclave_wrapper::get_instance().list_all_ids())
+    {
+        std::cout << id << "\n";
+    }
+}
+
```
Comment (tajriis) [inline untrusted/io_handler.cpp]: Output should be different when there is no item

File: untrusted/io_handler.cpp [hunk 2]
```diff
@@ -177,19 +180,41 @@ void io_handler::handle_remove_entry()
...
+    }
+
+    std::cout << "Listing all entries:\n";
+    std::cout << "===================================\n";
+    for (const auto& item : enclave_wrapper::get_instance().show_all_items())
+    {
+        std::cout << "Id: " << item.id << "\n";
+        std::cout << "Username: " << item.username << "\n";
+        std::cout << "Password: " << item.password << "\n";
+        std::cout << "===================================\n";
```
Comment (tajriis) [inline untrusted/io_handler.cpp]: Output should be different when there is no item

File: untrusted/io_handler.cpp [hunk 3]
```diff
@@ -77,35 +81,54 @@ int io_handler::run(std::size_t argc, char* argv[])
...
+        std::cout << e.what() << "\n";
         return 1;
     }
     return 0;
 }
 
-void io_handler::handle_help(const po::options_description& description_) noexcept
+void io_handler::handle_help(const po::options_description& description_)
 {
     std::cout << "Usage: options_description [options]\n";
```
Comment (JoyOfMadness) [inline untrusted/io_handler.cpp]:
```suggestion
    std::cout << "Usage: wuss [options]\n";
```

Comment (JoyOfMadness) [reply untrusted/io_handler.cpp]: Also, add some knid of description in a sense that this is `wuss` wallet and it stores passwords...

File: untrusted/io_handler.cpp [hunk 4]
```diff
@@ -116,75 +139,156 @@ void io_handler::handle_new_entry() noexcept
...
+    const std::string id = io_handler::get_item_id();
+    const auto old_item = enclave_wrapper::get_instance().show_item(id);
+    if (!old_item)
+    {
+        std::cout << "Entry with given id not found\n";
+        return;
+    }
+
+    item_t new_item;
+    new_item.id = gn("item", old_item->id);
```
Comment (JoyOfMadness) [inline untrusted/io_handler.cpp]: I think `ID` instead of `item` sould be more fitting

**Review by tajriis — CHANGES_REQUESTED** [review]

**Review by JoyOfMadness — CHANGES_REQUESTED** [review]
Few nitpicks

**Review by tajriis — APPROVED** [review]

File: untrusted/io_handler.cpp [hunk 5]
```diff
@@ -77,35 +81,56 @@ int io_handler::run(std::size_t argc, char* argv[])
...
     return 0;
 }
 
-void io_handler::handle_help(const po::options_description& description_) noexcept
+void io_handler::handle_help(const po::options_description& description_)
 {
-    std::cout << "Usage: options_description [options]\n";
+    std::cout << "Welcome to WUSS!\n";
+    std::cout << "WUSS enables you to securely store your passwords on all computers!\n";
+    std::cout << "Usage: ws [options]\n";
```
Comment (JoyOfMadness) [inline untrusted/io_handler.cpp]: typo, should be `wuss` instead of `ws`

**Review by JoyOfMadness — APPROVED** [review]

**Pull request merged**